lib/index.js

```diff
@@ -149,7 +151,7 @@ function getOptions(options) {
 
   options.error = function(err) {
     if (error) {
-      error(util._extend(new Error(), JSON.parse(err)));
+      error.call(options.context, util._extend(new Error(), JSON.parse(err)));
```

Comment:
I'm sorry @am11, I don't think I understand here. I wrote a test to see if anything was being dropped from the options object or we were somehow losing our function definitions.

```js
it('should preserve options by reference', function(done) {
  var options;
  options = {
    data: src,
    success: function() {
      assert.strictEqual(this.options, options);
      assert.strictEqual(this.options.importer, options.importer);
      done();
    },
    importer: function() {
      return {
        contents: 'div {color: yellow;}'
      };
    }
  };
  sass.render(options);
});
```

Because we are mutating objects by reference, everything seems to arrive as expected in the callbacks. Should I change something in the above test to capture your concerns? Thanks!